src/intptrcast.rs

```diff
@@ -163,6 +163,14 @@ impl<'mir, 'tcx> GlobalStateInner {
     }
 
     fn alloc_base_addr(ecx: &MiriInterpCx<'mir, 'tcx>, alloc_id: AllocId) -> u64 {
+        for (offset, chunk) in ecx.machine.mappings.all_memory.iter_all() {
+            if let Some(map) = chunk {
+                if map.alloc_id == alloc_id {
+                    return offset.bytes();
+                }
+            }
+        }
```

Comment:
I don't think I like having two tables for this... can't we have a single table for all things in our address space?